functions/transformations/min.go

```diff
@@ -131,33 +131,69 @@ func (s *MinSelector) selectRow(idx int, cr flux.ArrowColReader) {
 
 func (s *MinIntSelector) DoInt(vs *array.Int64, cr flux.ArrowColReader) {
 	minIdx := -1
-	for i, v := range vs.Int64Values() {
-		if !s.set || v < s.min {
-			s.set = true
-			s.min = v
-			minIdx = i
+	if vs.NullN() == 0 {
+		for i, v := range vs.Int64Values() {
+			if !s.set || v < s.min {
+				s.set = true
+				s.min = v
+				minIdx = i
+			}
+		}
+	} else {
+		for i := 0; i < vs.Len(); i++ {
+			if vs.IsValid(i) {
+				if v := vs.Value(i); !s.set || v < s.min {
+					s.set = true
+					s.min = v
+					minIdx = i
+				}
+			}
 		}
 	}
 	s.selectRow(minIdx, cr)
 }
 func (s *MinUIntSelector) DoUInt(vs *array.Uint64, cr flux.ArrowColReader) {
 	minIdx := -1
-	for i, v := range vs.Uint64Values() {
-		if !s.set || v < s.min {
-			s.set = true
-			s.min = v
-			minIdx = i
+	if vs.NullN() == 0 {
+		for i, v := range vs.Uint64Values() {
+			if !s.set || v < s.min {
+				s.set = true
+				s.min = v
+				minIdx = i
+			}
+		}
+	} else {
+		for i := 0; i < vs.Len(); i++ {
+			if vs.IsValid(i) {
+				if v := vs.Value(i); !s.set || v < s.min {
+					s.set = true
+					s.min = v
+					minIdx = i
+				}
+			}
 		}
 	}
 	s.selectRow(minIdx, cr)
 }
 func (s *MinFloatSelector) DoFloat(vs *array.Float64, cr flux.ArrowColReader) {
 	minIdx := -1
-	for i, v := range vs.Float64Values() {
-		if !s.set || v < s.min {
-			s.set = true
-			s.min = v
-			minIdx = i
+	if vs.NullN() == 0 {
+		for i, v := range vs.Float64Values() {
+			if !s.set || v < s.min {
+				s.set = true
+				s.min = v
+				minIdx = i
+			}
+		}
+	} else {
+		for i := 0; i < vs.Len(); i++ {
```

Comment:
Is there a meaningful difference between 

    for i, v := range vs.Float64Values()

and 

    for i := 0; i < vs.Len(); i++ {

Does is some how relate to how nulls are handled? If not then I suggest we use a consistent loop iteration mechanism.

Reply:
i don't know honestly. I assumed it might relate to function calls and whether it could be inlined or not. I'm not sure if that's a concern or not at the moment.

Reply:
My understanding is that the IsValid call should be inlined independent of the loop iteration method.

Reply:
Changed to only use the second method.